### PR TITLE
Elastishadow

### PR DESCRIPTION
**Pull request by bretambrose**
Adds elastishadow command line test utility.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: bin/elastishadow/main.c [hunk 1]
```diff
@@ -0,0 +1,1261 @@
...
+    struct aws_tls_connection_options tls_connection_options;
+
+    const char *log_filename;
+    enum aws_log_level log_level;
+
+    struct aws_mqtt5_client *client;
+    struct aws_mqtt_request_response_client *rr_client;
+
+    /*
+     * &aws_shadow_streaming_operation.id -> aws_shadow_streaming_operation *
```
Comment (sbSteveK) [inline bin/elastishadow/main.c:57]: Intentional comment?

File: bin/elastishadow/main.c [hunk 2]
```diff
@@ -0,0 +1,1261 @@
...
+}
+
+void s_aws_shadow_streaming_operation_destroy(struct aws_shadow_streaming_operation *operation) {
+    if (operation == NULL) {
+        return;
+    }
+
+    aws_byte_buf_clean_up(&operation->thing);
+    aws_byte_buf_clean_up(&operation->shadow);
+
```
Comment (sbSteveK) [inline bin/elastishadow/main.c:101]: May be intentional or handled elsewhere later but noting now that we're not cleaning up the topic_filter byte buf in this destroy.

File: bin/elastishadow/main.c [hunk 3]
```diff
@@ -0,0 +1,1261 @@
...
+    };
+
+    printf(
+        "Submitting GetNamedShadow request for shadow '" PRInSTR "' of thing '" PRInSTR
+        "' using correlation token %s...\n",
+        AWS_BYTE_CURSOR_PRI(shadow_name_cursor),
+        AWS_BYTE_CURSOR_PRI(thing_name_cursor),
+        correlation_token);
+
+    if (aws_mqtt_request_response_client_submit_request(context->rr_client, &get_options) == AWS_OP_ERR) {
```
Comment (sbSteveK) [inline bin/elastishadow/main.c:341]: Trivial: We should `aws_string_destroy` the correlation_token here if the completion isn't going to be called.

File: bin/elastishadow/main.c [hunk 4]
```diff
@@ -0,0 +1,1261 @@
...
+    aws_array_list_get_at(&words, &command_cursor, 0);
+
+    if (aws_byte_cursor_eq_ignore_case(&command_cursor, &quit_cursor)) {
+        printf("Quitting!\n");
+        done = true;
+    } else if (aws_byte_cursor_eq_ignore_case(&command_cursor, &start_cursor)) {
+        printf("Starting client!\n");
+        aws_mqtt5_client_start(client);
+    } else if (aws_byte_cursor_eq_ignore_case(&command_cursor, &stop_cursor)) {
+        aws_mqtt5_client_stop(client, NULL, NULL);
```
Comment (sbSteveK) [inline bin/elastishadow/main.c:971]: Trivial: printf("Stopping client!\n") ?

File: bin/elastishadow/main.c (hunk 3 above)
Comment (sbSteveK) [reply bin/elastishadow/main.c:341]: If applicable here, applicable in all places where we check the result of `aws_mqtt_request_response_client_submit_request`

**Review by sbSteveK — APPROVED** [review]
Looks good! Not necessary but might be nice to also allow arg setting of max subscription options to manually test blocked and no capacity.

File: bin/elastishadow/main.c (hunk 2 above)
Comment (bretambrose) [reply bin/elastishadow/main.c:101]: Yes, that was missed.

**Pull request closed without merging**